students/aminavi/session04/mailroom2.py

```diff
@@ -0,0 +1,80 @@
+import sys
+import tempfile
+
+donor_list = {
+    'william': [653772.32 , 953772.32,100000],
+    'jeff': [8888777.22 ,877.33],
+    'paul': [663.23, 43.87 ,1.32],
+    'mark': [1663.23, 4300.87, 10432.00],
+    'nigel': [5000000 , 10]
+    }
+
+
+def send_thank_you():
+    name = ''
+    while not name or name.lower() == 'list':
+        name = input("Please enter the full name: ")
+        if name.lower() == 'list':2
+        donor_list.setdefault(name.lower(),[])
+    donation = int(input("Please enter the donation amount: "))
+    for i in range(len(donor_list.keys())):
```

Comment:
Why a loop here? Because of this loop, your line 21-22 will be executed `len(donor_list.keys())` times, which means the one donation amount `donation` will be appended that many times, which I don't think is right...? Is that what you intended?